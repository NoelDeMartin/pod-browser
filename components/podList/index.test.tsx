import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

import PodList from "./index";

describe("Pod list", () => {
  test("Renders null if there are no pod iris", () => {
    const iris = undefined;
    const tree = shallow(<PodList podIris={iris} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  test("Renders a container if there is one pod iri", () => {
    const iris = ["https://mypod.myhost.com"];
    const tree = shallow(<PodList podIris={iris} />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  test("Renders a table if there are multiple pod iris", () => {
    const iris = [
      "https://mypod.myhost.com",
      "https://myotherpod.myotherhost.com",
    ];

    const tree = shallow(<PodList podIris={iris} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
