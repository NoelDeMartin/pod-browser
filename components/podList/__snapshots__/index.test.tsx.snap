// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pod list Renders a container if there is one pod iri 1`] = `
<Container
  iri="https://mypod.myhost.com"
/>
`;

exports[`Pod list Renders a table if there are multiple pod iris 1`] = `
<WithStyles(ForwardRef(Table))>
  <WithStyles(ForwardRef(TableHead))>
    <WithStyles(ForwardRef(TableRow))>
      <WithStyles(ForwardRef(TableCell))>
        Pod IRI
      </WithStyles(ForwardRef(TableCell))>
    </WithStyles(ForwardRef(TableRow))>
  </WithStyles(ForwardRef(TableHead))>
  <WithStyles(ForwardRef(TableBody))>
    <WithStyles(ForwardRef(TableRow))
      key="https://mypod.myhost.com"
    >
      <WithStyles(ForwardRef(TableCell))>
        <a
          href="/resource/https://mypod.myhost.com"
        >
          https://mypod.myhost.com
        </a>
      </WithStyles(ForwardRef(TableCell))>
    </WithStyles(ForwardRef(TableRow))>
    <WithStyles(ForwardRef(TableRow))
      key="https://myotherpod.myotherhost.com"
    >
      <WithStyles(ForwardRef(TableCell))>
        <a
          href="/resource/https://myotherpod.myotherhost.com"
        >
          https://myotherpod.myotherhost.com
        </a>
      </WithStyles(ForwardRef(TableCell))>
    </WithStyles(ForwardRef(TableRow))>
  </WithStyles(ForwardRef(TableBody))>
</WithStyles(ForwardRef(Table))>
`;

exports[`Pod list Renders null if there are no pod iris 1`] = `""`;
